### ISSUE TITLE
[Feat]: Retention for statsd metrics

### ISSUE THREAD

**Issue by victorvoronin** [netdata/netdata#16149]
### Problem

We use netdata statsd to collect custom statsd metrics from containers and send them to Graphite. Metric names include container name that is why we have continuous private charts number (statsd metrics) grow. But those metrics are short lived. We want to have some sort of retention for private charts, because without it we have to restart netdata after reaching "max private charts hard limit".

### Description

At the moment we only can set "max private charts hard limit" for in [statsd] conf file section. This means that when you reach this limits you will not be able to add more private charts (statsd metrics).
We want to be able to set private charts retention period to avoid high memory usage by private charts and to avoid netdata restarts.

### Importance

really want

### Value proposition

1. Avoid high memory usage in case of huge number of private charts
2. Avoid netdata custom restarts after reaching "max private charts hard limit"

**Comment by MrZammler** [comment]
Hi @victorvoronin 

Sorry for the late reply. We uploaded a PR to try and address this. Basically you can set a time after which any metrics that have stopped being send, will be set to obsolete (and cleanup). Would something like this address your issue?

**Comment by victorvoronin** [comment]
> Hi @victorvoronin
> 
> Sorry for the late reply. We uploaded a PR to try and address this. Basically you can set a time after which any metrics that have stopped being send, will be set to obsolete (and cleanup). Would something like this address your issue?

@MrZammler Sorry, I've missed your message. Looks like this should address our issue. I'll test it as soon as possible

**Comment by victorvoronin** [comment]
@MrZammler I've tested version with your commit and looks like our issue is not solved.
I use "set charts as obsolete after secs" parameter and see that private charts number is not only growing. But, on the other hand I see that another parameter "cleanup obsolete charts after secs" is not implemented in the code. As a result I see that memory usage of netdata process is still growing in time, like if private charts data of obsolete charts still live in memory.
So, to recap, I see that something is changed, but memory usage is still not as expected.

**Comment by MrZammler** [comment]
@victorvoronin thanks for the feedback.

We will check the `cleanup obsolete charts after secs` behavior. (One thing to note though, is that obsolete charts won't be deleted if they are being displayed (e.g. appear on the dashboard) but I guess they would eventually be deleted during e.g. night time...)

**Comment by victorvoronin** [comment]
@MrZammler Any news on my question?
I've checked your suggestion about deletion during night time, and it doesn't work. Memory consuption grows until netdata restarted
